base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
-// TODO C Destructuring methods for the KeyManagerStateFfi struct
-
-#[no_mangle]
-pub unsafe extern "C" fn get_seed_words(wallet: *mut WalletFfi) -> *mut KeyManagerSeedWords {}
-// TODO C Destructuring methods for the KeyManagerSeedWords struct
+//#[no_mangle]
+// pub unsafe extern "C" fn generate_key_manager_from_seed_words(
+// wallet: *mut Wallet,
+//    seed_words: *mut KeyManagerSeedWords,
+//    branch_seed: *const c_char, // String
+//) -> bool
+//{
+//}
 
-#[no_mangle]
-pub unsafe extern "C" fn create_seed_words() -> KeyManagerSeedWords {}
-
-#[no_mangle]
-pub unsafe extern "C" fn add_seed_word(
-    seed_words: *mut KeyManagerSeedWords,
-    word: *const c_char, // String
-) -> bool
-{
-}
+//#[no_mangle]
+// pub unsafe extern "C" fn generate_identity(wallet: *mut Wallet) -> *mut IdentityFfi {}
+// TODO C Destructuring methods for the IdentityFfi struct
 
 #[no_mangle]
-pub unsafe extern "C" fn generate_key_manager_from_seed_words(
-    wallet: *mut WalletFfi,
-    seed_words: *mut KeyManagerSeedWords,
-    branch_seed: *const c_char, // String
-) -> bool
-{
-}
+pub unsafe extern "C" fn wallet_add_base_node_peer(wallet: *mut Wallet, peer: *mut Peer) -> bool {
+    if wallet.is_null() {
+        return false;
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn generate_identity(wallet: *mut WalletFfi) -> *mut IdentityFfi {}
-// TODO C Destructuring methods for the IdentityFfi struct
+    if peer.is_null() {
+        return false;
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn add_base_node_peer(
-    wallet: *mut WalletFfi,
-    public_key: *const c_char,  // Byte[32] - PublicKey
-    secret_key: *const c_char,  // Byte[32] - SecretKey
-    net_address: *const c_char, // NetAddress
-) -> bool
-{
+    // (*wallet).addPeer((*peer));
+    return true;
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn get_network_status(wallet: *mut WalletFfi) -> *mut NetworkStatusFfi {}
+//#[no_mangle]
+// pub unsafe extern "C" fn get_network_status(wallet: *mut Wallet) -> *mut NetworkStatusFfi {}
 // TODO C Destructuring methods for the NetworkStatusFfi struct
 
 #[no_mangle]
-pub unsafe extern "C" fn get_balance(wallet: *mut WalletFfi) -> c_ulonglong {}
+pub unsafe extern "C" fn wallet_get_balance(wallet: *mut Wallet) -> c_ulonglong {
+    //(*wallet).getBalance();
+    return 0;
+}
 
 // Create and send the first stage of a transaction to the specified wallet for the specified amount and with the
 // specified fee.
 #[no_mangle]
-pub unsafe extern "C" fn send_transaction(
-    wallet: *mut WalletFfi,
-    amount: c_ulonglong,       // MicroTari
-    fee_per_gram: c_ulonglong, // MicroTari
-    lock_height: c_ulonglong,  // u64
-    // Destination Node Peer
-    public_key: *const c_char,  // Byte[32] - PublicKey
-    net_address: *const c_char, // NetAddress
+pub unsafe extern "C" fn wallet_send_transaction(
+    wallet: *mut Wallet,
+    peer: *mut Peer,
+    transaction: *mut Transaction,
```

Comment:
The Transaction type is a completed transaction. This method call starts the process of negotiation. The parameters I specified are the arguments required for this process to begin.